### PR TITLE
Add user models

### PR DESCRIPTION
**Pull request by DeepikaGhodki**
https://github.com/microbiomedata/nmdc-server/issues/689

1. Create user if not already present at login
2. Refer to user id from submission metadata (replace orcid in future)
3. Migration: populate all users already present in submission_metatdata, bulk_download and file_download

### CODE REVIEW

File: nmdc_server/schemas.py [hunk 1]
```diff
@@ -657,6 +657,25 @@ class SubmissionMetadataSchema(SubmissionMetadataSchemaCreate):
     author_orcid: str
     created: datetime
     status: str
+    user: Optional[User]
```
Comment (subdavis) [inline nmdc_server/schemas.py:660]: Change to `author`.

**Review by subdavis — CHANGES_REQUESTED** [review]
Code looks great!  I have a few mostly minor comments.  Once these changes have been addressed, I will download and run this PR.

**Comment by turbomam** [comment]
This is great. We will need to synchronize any modeling here with the nmdc-schema

File: nmdc_server/auth.py [hunk 2]
```diff
@@ -85,10 +89,18 @@ async def login_required(token: Optional[Token] = Depends(get_token)) -> Token:
...
+) -> Token:
     if settings.environment != "production":
         return token
-    if token.orcid in _admin_users:
+    user = crud.get_user(db, token.orcid)
+    if user is None:
+        raise HTTPException(
+            status_code=status.HTTP_403_FORBIDDEN,
+            detail="No user found",
+        )
```
[the remark this answers is not in the document]
Comment (DeepikaGhodki) [reply nmdc_server/auth.py:102]: Ideally, we should never run into this as every user who is able to reach here would have logged in and hence user should be always found. And now that we are handling the user addition in migration, we will never encounter a scenario where there's a submission for user not present in the system. 

My thought process here was that irrespective of how the user got here, if we can't validate we consider it as an invalid case. If this happens with a valid user, then they may login again. I see your point through, 403 could confuse the user. 
If we are propagating the error message, we can add an actionable message. 

Or we could let it fail with an unexpected error i.e. without raising any exception. But what would the user do then? Reload the page, and retry maybe?

**Comment by turbomam** [comment]
Is there any documentation for this, or a non-code serialization of the model?

**Pull request merged**